model/bindings.go

```diff
@@ -0,0 +1,122 @@
+func CreateBindingsRancher(relativePath string) map[string]BindingProperty {
+	var rawConfigRancher config.RawServiceMap
+	var rancherComposeMap map[string]rancher.RancherConfig
+	var bindingsMap map[string]BindingProperty
+
+	var configMap map[string]config.ServiceConfigV1
+	var rawConfigDocker config.RawServiceMap
+
+	bindingsMap = make(map[string]BindingProperty)
+
+	rancherFile := relativePath + "/rancher-compose.yml"
+
+	dockerFile := relativePath + "/docker-compose.yml"
+	fileExists, err := os.Stat(dockerFile)
+	_ = fileExists
+	if err == nil {
+		yamlContent, err := ioutil.ReadFile(dockerFile)
+		if err != nil {
+			log.Errorf("Error in opening file : %v\n", err)
+		} else {
+			err = yaml.Unmarshal(yamlContent, &rawConfigDocker)
+			if err != nil {
+				log.Debugf("Error during Unmarshal for file %s : %v\n", dockerFile, err)
+			} else {
+				rawConfigDocker, err := preprocess.PreprocessServiceMap(rawConfigDocker)
+				utils.Convert(rawConfigDocker, &configMap)
+				_ = rawConfigDocker
+				if err != nil {
+					log.Debugf("Error")
+				}
+			}
+		}
+	}
+
+	rancherYaml, err := ioutil.ReadFile(rancherFile)
+	if err != nil {
+		log.Errorf("Error in opening file : %v\n", err)
+		return nil
+	}
+
+	err = yaml.Unmarshal(rancherYaml, &rawConfigRancher)
+	if err != nil {
+		log.Errorf("Error during Unmarshal  for rancher-compose: %v\n", err)
+	} else {
+		rawConfigRancher, err := preprocess.PreprocessServiceMap(rawConfigRancher)
+		utils.Convert(rawConfigRancher, &rancherComposeMap)
+		if err != nil {
+			fmt.Printf("Error")
+		} else {
+			keys := reflect.ValueOf(rancherComposeMap).MapKeys()
+			for _, key := range keys {
+				if key.String() != ".catalog" {
+					newBinding := BindingProperty{}
+					newBinding.Binding.Labels = mapLabel{}
+					newBinding.Binding.Ports = []string{}
+
+					newBinding.Binding.Scale = rancherComposeMap[key.String()].Scale
+					if configMap[key.String()].Labels != nil {
+						newBinding.Binding.Labels = configMap[key.String()].Labels
+					}
+					if configMap[key.String()].Ports != nil {
+						newBinding.Binding.Ports = configMap[key.String()].Ports
+					}
+					bindingsMap[key.String()] = newBinding
+				}
+			}
+
+			keys = reflect.ValueOf(configMap).MapKeys()
+			for _, key := range keys {
+				if val, serviceParsed := bindingsMap[key.String()]; serviceParsed {
+					log.Debugf("Service bindings already provided")
+					_ = val
+					continue
+				}
+				if key.String() != ".catalog" {
+					newBinding := BindingProperty{}
+					newBinding.Binding.Labels = mapLabel{}
+					newBinding.Binding.Ports = []string{}
+					if configMap[key.String()].Labels != nil {
+						newBinding.Binding.Labels = configMap[key.String()].Labels
+					}
+					if configMap[key.String()].Ports != nil {
+						newBinding.Binding.Ports = configMap[key.String()].Ports
+					}
+					bindingsMap[key.String()] = newBinding
+				}
+			}
+			return bindingsMap
+		}
+	}
+	return nil
+}
```

Comment:
I just got this checked with Josh that ports and labels are only in docker-compose.yml file. Scale can be in both and if it is then the one in rancher-compose wins